lib/fencing/st_client.c

```diff
@@ -2537,13 +2552,38 @@ stonith__event_description(stonith_event_t *event)
         status = crm_exit_str(CRM_EX_OK);
     }
 
+    if (pcmk__str_eq(event->operation, T_STONITH_NOTIFY_HISTORY,
+                     pcmk__str_none)) {
+        return crm_strdup_printf("Fencing history may have changed");
+
+    } else if (pcmk__str_eq(event->operation, STONITH_OP_DEVICE_ADD,
+                            pcmk__str_none)) {
+        return crm_strdup_printf("A fencing device (%s) was added", device);
+
+    } else if (pcmk__str_eq(event->operation, STONITH_OP_DEVICE_DEL,
+                            pcmk__str_none)) {
+        return crm_strdup_printf("A fencing device (%s) was removed", device);
+
+    } else if (pcmk__str_eq(event->operation, STONITH_OP_LEVEL_ADD,
+                            pcmk__str_none)) {
+        return crm_strdup_printf("A fencing topology level (%s) was added",
+                                 device);
+
+    } else if (pcmk__str_eq(event->operation, STONITH_OP_LEVEL_DEL,
+                            pcmk__str_none)) {
+        return crm_strdup_printf("A fencing topology level (%s) was removed",
+                                 device);
+    }
+
+    // event->operation should be T_STONITH_NOTIFY_FENCE
+
     return crm_strdup_printf("Operation %s of %s by %s for %s@%s: %s%s%s%s (ref=%s)",
-                             event->action, event->target, executioner, origin,
-                             event->origin, status,
+                             action, target, executioner, origin, origin_node,
+                             status,
                              ((reason == NULL)? "" : " ("),
                              ((reason == NULL)? "" : reason),
                              ((reason == NULL)? "" : ")"),
-                             event->id);
+                             ((event->id == NULL)? "(none)" : event->id));
```

Comment:
well we're using our assignment just once anyway but for consistency we could ad id to the default-assignment above.